File: DraftRightMobile/android/app/src/main/kotlin/com/draftright/keyboard/lang/ItalianLanguagePack.kt
package com.draftright.keyboard.lang

import com.draftright.keyboard.KeyDef
import com.draftright.keyboard.LanguagePack
import java.util.Locale

object ItalianLanguagePack : LanguagePack {
    override val id = "it"
    override val displayName = "Italiano"
    override val locale: Locale = Locale.ITALIAN

    override val alphaRows: List<List<KeyDef>> = EnglishLanguagePack.alphaRows
    override val symbols1Rows: List<List<KeyDef>> = EnglishLanguagePack.symbols1Rows
    override val symbols2Rows: List<List<KeyDef>> = EnglishLanguagePack.symbols2Rows

    override val longPressAccents: Map<Char, List<Char>> = mapOf(
        'a' to listOf('à', 'á', 'â', 'ã', 'ä'),
        'e' to listOf('è', 'é', 'ê', 'ë'),
        'i' to listOf('ì', 'í', 'î', 'ï'),
        'o' to listOf('ò', 'ó', 'ô', 'õ', 'ö'),
        'u' to listOf('ù', 'ú', 'û', 'ü'),
    )
}

File: DraftRightMobile/ios/DraftRightKeyboardCore/Sources/DraftRightKeyboardCore/Lang/ItalianLanguagePack.swift
import Foundation

public struct ItalianLanguagePack: LanguagePack {
    public let id = "it"
    public let displayName = "Italiano"
    public let locale = Locale(identifier: "it")

    private let base = EnglishLanguagePack()

    public init() {}

    public var alphaRows: [[KeyDef]] { base.alphaRows }
    public var symbols1Rows: [[KeyDef]] { base.symbols1Rows }
    public var symbols2Rows: [[KeyDef]] { base.symbols2Rows }

    public let longPressAccents: [Character: [Character]] = [
        "a": ["à", "á", "â", "ã", "ä"],
        "e": ["è", "é", "ê", "ë"],
        "i": ["ì", "í", "î", "ï"],
        "o": ["ò", "ó", "ô", "õ", "ö"],
        "u": ["ù", "ú", "û", "ü"],
    ]
}
